sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -423,25 +444,45 @@ def refresh(self, **kwargs) -> None:
-                        label=label,
-                        etag=etag,
-                        match_condition=MatchConditions.IfModified,
-                        headers=_get_headers("Watch", uses_key_vault=self._uses_key_vault, **kwargs),
-                        **kwargs
-                    )
-                    if updated_sentinel is not None:
-                        logging.debug(
-                            "Refresh all triggered by key: %s label %s.",
-                            key,
-                            label,
+                need_refresh = False
+                updated_sentinel_keys = dict(self._refresh_on)
+                headers = _get_headers("Watch")
+                for (key, label), etag in updated_sentinel_keys.items():
+                    try:
+                        updated_sentinel = self._client.get_configuration_setting(
+                            key=key,
+                            label=label,
+                            etag=etag,
+                            match_condition=MatchConditions.IfModified,
+                            headers=headers,
+                            **kwargs
                         )
-                        self._load_all(headers=_get_headers("Watch", uses_key_vault=self._uses_key_vault, **kwargs))
-                        self._refresh_on[(key, label)] = updated_sentinel.etag
-                        self._refresh_timer.reset()
-                        return
+                        if updated_sentinel is not None:
+                            logging.debug(
+                                "Refresh all triggered by key: %s label %s.",
+                                key,
+                                label,
+                            )
+                            need_refresh = True
+
+                            updated_sentinel_keys[(key, label)] = updated_sentinel.etag
+                    except HttpResponseError as e:
+                        if e.status_code == 404:
+                            if etag is not None:
+                                # If the sentinel is not found, it means the key/label was deleted, so we should refresh
+                                logging.debug("Refresh all triggered by key: %s label %s.", key, label)
+                                need_refresh = True
+                                updated_sentinel_keys[(key, label)] = None
+                        else:
+                            raise e
+                # Need to only update once, no matter how many sentinels are updated
+                if need_refresh:
+                    self._load_all(sentinel_keys=updated_sentinel_keys, **kwargs)
+                    if self._on_refresh_success:
+                        self._on_refresh_success()
+                # Even if we don't need to refresh, we should reset the timer
+                self._refresh_timer.reset()
+                return
         except (ServiceRequestError, ServiceResponseError) as e:
```

Comment:
What is the reason for checking `_is_retryable_error` and swallowing the exception?

Reply:
Side note: 502 and 503 are also retriable errors.
https://github.com/Azure/azure-sdk-for-python/blob/7e152d8139e84ff761497ccf0e8fbe72afdaec8e/sdk/core/azure-core/azure/core/pipeline/policies/_retry.py#L73C67-L73C67

Reply:
If it is retriable, then in theory, there is no action for them besides calling backoff again in a shorter amount of time.